manager/controllers/app/m4dapplication_controller.go

```diff
@@ -272,3 +268,22 @@ func (r *M4DApplicationReconciler) GetAllModules() (map[string]*app.M4DModule, e
 	}
 	return moduleMap, nil
 }
+
+// CreateAppNamespace creates a namespace in which the blueprint and the relevant resources will be running
+// It returns the generated namespace name
+func (r *M4DApplicationReconciler) CreateAppNamespace(app *app.M4DApplication) (string, error) {
+	genNamespace := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{GenerateName: "m4d-"}}
```

Comment:
Would be useful (for debugging) to add a label with the value of the application name

Reply:
Agree with @roee88 that for debugging and logging purposes we need some way to see an association with the M4DApplication

Reply:
Added labels to the generated namespace.